src/game_api/script/lua_vm.cpp

```diff
@@ -680,12 +680,14 @@ end
     /// Show a message that looks like a level feeling.
     lua["toast"] = [](std::u16string message)
     {
+        State::get().ptr()->toast = 0;
```

Comment:
Won't this cause memory leak?
also it changes the behavior, for example the `say` function have some sort of buffer, so if you call it couple of times the second message will show up after the first one disappears

Reply:
Oh I assumed you'd always want to display a new message with these and the one put on the screen by the game was just an annoyance. At least that's probably the case with toast. The cancel_functions should still be fixed so they can be used on the same frame, but `delete` didn't fix the memory leak either. Reverted anyway.

Reply:
You know

```lua
toast("some boring vanilla feeling set by the vanilla level gen")
cancel_toast() -- should cancel immeditely, i.e. set the toast to null so toast() can be used again immediately
toast("custom feeling should always override vanilla feeling, at least after cancel_toast, without waiting for next update")
```